scripts/ci-e2e.sh

```diff
@@ -53,7 +53,12 @@ export USE_EXISTING_CLUSTER=false
 # - KUBERNETES_VERSION_UPGRADE_FROM
 # - KUBERNETES_VERSION_LATEST_CI
 # - KUBERNETES_VERSION_MANAGEMENT
-k8s::prepareKindestImagesVariables
+if [[ -n "${GINKGO_LABEL_FILTER+x}" && -n "${GINKGO_LABEL_FILTER}" ]]; then
```

Comment:
I wonder if we should simply add GINKGO_LABEL_FILTER first in test-infra so we don't have to handle both 

(I would also add GINKGO_LABEL_FILTER in test-infra on all branches, do avoid if/else in templates for now)

Reply:
definitely open to changing the approach here -- was thinking it would be "safest" to do it sort of incrementally, but it does introduce complexity given how this is currently configured 😅

Reply:
Let's see how https://github.com/kubernetes-sigs/cluster-api/pull/11763#discussion_r1936007717 turns out

Reply:
My suggestion was also to do it incrementally. Just basically starting with adding GINKGO_LABEL_FILTER env vars additioanlly in test-infra first :)

Reply:
Let's forget this thread and restart the discussion based on what we want to do now that everything is simpler :D